packages/shared/src/components/modals/BootPopups.tsx

```diff
@@ -242,6 +244,30 @@ export const BootPopups = (): ReactElement => {
     user,
   ]);
 
+  /**
+   * Received gift plus modal
+   */
+  useEffect(() => {
+    if (!user?.shouldShowGiftPlus || !user.isPlus) {
+      return;
+    }
+
+    addBootPopup({
+      type: LazyModal.ReceivedGiftPlus,
+      props: {
+        user,
+        onAfterClose: () => {
+          updateUserProfile({
+            flags: { showPlusGift: false },
+          });
```

Comment:
Do we have to wait for the user to close it before we turn off the flag?

Reply:
I don't think this is mandatory to wait plus don't return any promise. If you think is important I can change this behaviour 👀

Reply:
Maybe it is more of how we should be doing it generally for our modals. Anyway, shouldn't be a blocker and we can likely keep it.